charts/consul/values.yaml

```
@@ -114,6 +114,10 @@ global:
   # secretKey should be in the form of "key".
   secretsBackend:
     vault:
+      # Vault namespace (optional).
+      # For more details, please refer to [Vault Connect CA configuration](https://developer.hashicorp.com/consul/docs/connect/ca/vault#namespace)
```

Since we're going w/ the higher-level config key, we should probably update this to also note the annotation it controls.
```suggestion
      # Vault namespace (optional). This sets the Vault namespace for the `vault.hashicorp.com/namespace` 
      # agent annotation and [Vault Connect CA configuration](https://developer.hashicorp.com/consul/docs/connect/ca/vault#namespace).
      # To override one of these values individually, see `agentAnnotations` and `connectCA.additionalConfig`.
```
@david-yu @absolutelightning does this look about right to you?

Also thinking we should test that both of these fields take precedence over the higher-level one we're adding when both are specified, if that's a use case we intend to support.

---

@zalimeni Adding namespace to connectCA.additionalConfig will not update the annotation. It will just override the namespace. also Additional config is now not the preferred way so I had removed test for additional config.
Currently - higher-level one if present will add namespace to connectCA and create vault namespace annotation.
If annotations are already provided by user it will not append vault namespace annotation.
cc @david-yu is this correct?

---

> @zalimeni Adding namespace to connectCA.additionalConfig will not update the annotation. It will just override the namespace. also Additional config is now not the preferred way so I had removed test for additional config.
> Currently - higher-level one if present will add namespace to connectCA and create vault namespace annotation.
> If annotations are already provided by user it will not append vault namespace annotation.
> cc @david-yu is this correct?

Yes that is what we should do. The docs do look correct to me.

---

> Adding namespace to connectCA.additionalConfig will not update the annotation. It will just override the namespace.

Yep, that makes sense 👍🏻 what I was trying to clarify in the docs is that there are two individual values that can be separately overridden, regardless of whether the top-level field is set.

> Additional config is now not the preferred way so I had removed test for additional config.

Based on updated docs proposed, I think that we'd want two specific test cases covered:
- if `vault.vaultNamespace` and `agentAnnotations` both specify a namespace, the `agentAnnotations` value will be used for the annotation
- if `vault.vaultNamespace` and `connectCA.additionalConfig` both specify a namespace, the `connectCA.additionalConfig` value will be used for the config

---

ok yes.

---

I have updated bats tests for point 1. but point 2 I am not able to achieve, not sure if bats accepts json in --set flag.
Can you please help. @zalimeni

---

there is a -set-json flag but I think this will not be in bats tests. rather go tests.

---

@zalimeni Point 1 is covered using bats tests.
Regarding Point 2 - do we want to keep this way of setting namespace in our codebase where we use additionalConfig to set namespace - we removed the comment [here](https://github.com/hashicorp/consul-k8s/pull/2841#discussion_r1310434027). Also i feel adding on more go test will increase CI time and since additionalConfig is no longer a preferred way we can exclude that. What are your thoughts?

---

@absolutelightning I'm not sure how much of a priority it is to preserve the option of using `additionalConfig` namespace as an override for the new KV - I'm more just thinking if we document it, we should confirm it works. An alternative would be not to document it for now, if it's difficult to test and we're not sure it's going to be needed by customers. Maybe @david-yu has thoughts?

Re: docs (if we do keep it), I don't feel strongly about removing the example, so happy to defer to you and David on that. I agree it's more nuanced in the override case; it was a simpler decision when we were just adding an explicit field for `connectCA`.